DatadogCore/Tests/Datadog/DatadogCore/Context/CarrierInfoPublisherTests.swift

```diff
@@ -45,6 +45,7 @@ class CarrierInfoPublisherTests: XCTestCase {
         }
     }
 
+    @available(iOS, deprecated: 12)
     func testGivenCellularServiceUnavailableOnIOS11_whenReadingCurrentCarrierInfo_itReturnsNoValue() {
         // Given
         let reader = iOS11CarrierInfoReader(networkInfo: unavailableCTTelephonyNetworkInfo)
```

Comment:
Needed to deprecate it as we're using `iOS11CarrierInfoReader` here, deprecated on iOS 12. If not deprecating the test, it was causing build warnings in Xcode 15 (`IPHONEOS_DEPLOYMENT_TARGET=12.0`) and because we treat them as errors → build failure.